File: frontend/src/types/activity.ts
import type { ProjectStage } from '@/types/project'

export interface ActivityItem {
  id: string
  project_id: string
  project_code: string
  project_name: string
  stage: ProjectStage
  actor_role: 'rs' | 'vendor' | 'bank_mandiri' | null
  note: string | null
  created_at: string
  is_cross_sell_opportunity: boolean
}

export interface ActivityFeed {
  unread_count: number
  items: ActivityItem[]
}

File: backend/src/app/service_layer/schemas/activity.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.adapters.db.models.enums import ProjectStage, UserRole


class ActivityItem(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    project_code: str
    project_name: str
    stage: ProjectStage
    actor_role: UserRole | None
    note: str | None
    created_at: datetime
    is_cross_sell_opportunity: bool = False


class ActivityFeed(BaseModel):
    unread_count: int
    items: list[ActivityItem]
